app/javascript/application.js

```diff
@@ -48,6 +48,18 @@ import 'jquery-ui/themes/base/sortable.css'
 import { I18n } from "i18n-js";
 import locales from "../../tmp/locales.json";
 
+// create a promise that only gets resolved when turbolinks:load fired and all select2 locales have been loaded
+let select2_locales_loaded = new Promise((resolve, reject) => {
+  document.addEventListener('turbolinks:load', resolve);
+});
+Object.keys(locales).forEach(locale =>
+    select2_locales_loaded = Promise.all([select2_locales_loaded, import(`select2/dist/js/i18n/${locale}`)])
+);
+window.select2_locales_loaded = select2_locales_loaded;
```

Comment:
Constructing multiple promises here (just by merging two promises together in each iteration) seems to be a bit complicated. Do you think we can simplify that a little?

```js
Promise.all(
  Object.keys(locales).map(locale => import(`select2/dist/js/i18n/${locale}`))
).then(() =>
  window.dispatchEvent(new Event("select2_locales_loaded"))
);
```

Reply:
That looks much better, thanks.